### DISCUSSION TITLE
Error decoding JWT

### DISCUSSION THREAD

**Discussion by talan-z** [dani-garcia/vaultwarden#1786] [Ideas]
Hello!
I have below issue. Hopefully there is an easy fix!

### Subject of the issue
I created a new organization and invited a friend to join. Upon clicking the link, filling up the new account information, and submitting, the following appears on the webpage: Error decoding JWT.

### Your environment
* Bitwarden_rs version: 1.16.0
* Install method: docker
* Clients used: Webpage
* Reverse proxy and version: caddy v1.0.5
* Version of mysql/postgresql: n/a
* Other relevant information: 

### Relevant logs
bitwarden_1  | [2020-07-26 14:12:36][request][INFO] GET /api/organizations/a52078e7-0de5-455c-93f9-872937062640/policies/token?token=eyJ0eXAiOiJKV1QiLCJhbGci
bitwarden_1  | [2020-07-26 14:12:36][error][ERROR] Error decoding JWT.
bitwarden_1  | [CAUSE] Error(
bitwarden_1  |     ExpiredSignature,
bitwarden_1  | )
bitwarden_1  | [2020-07-26 14:12:36][response][INFO] GET /api/organizations/<org_id>/policies/token?<token> (list_policies_token) => 400 Bad Request
bitwarden_1  | [2020-07-26 14:12:50][request][INFO] POST /api/accounts/register
bitwarden_1  | [2020-07-26 14:12:50][error][ERROR] Error decoding JWT.
bitwarden_1  | [CAUSE] Error(
bitwarden_1  |     ExpiredSignature,
bitwarden_1  | )
bitwarden_1  | [2020-07-26 14:12:50][response][INFO] POST /api/accounts/register (register) => 400 Bad Request

Thank you,
Philipp

**Comment by dani-garcia** [comment]
The emails expire after five days I think, did he open the link after that time has passed?

**Comment by BlackDex** [comment]
Closing this long open issue. Please feel free to reopen if it is still not resolved 😄.

**Comment by ItsShadowCone** [comment]
As of currently, this is still the case. Also the email does not say anything about the five day limit.

We should provide a different error message if token decoding fails due to expiration, especially for invite links.

**Comment by BlackDex** [comment]
Well, the fefault value is 5 for upstream (although you can configure it there if self-hosted).
But it also does not mention this limit anywhere as far i can remember.

Though, we maybe can try to change the message maybe.

**Comment by BlackDex** [comment]
I'm going to close this again, as we want to try and keep as close as possible to upstream.
If someone really want's to implement something regarding this we may accept a well written PR.

But in my opinion, it will only add more complexity since all tokens use the same decoding function.
